kolibri/plugins/registry.py

```diff
@@ -1,21 +1,41 @@
 """
-Plugins are initialized once when the server starts. Here are the initilization steps:
+How plugins work
+----------------
 
-1. Each module listed in INSTALLED_APPS (including those listed in the kolibri conf file) is searched for the special
-   ``kolibri_plugin`` module.
-2. For each class listed in the module's ``PLUGINS`` property, which inherits from ``KolibriPluginBase``, it's ``hooks``
-   method is called to inspect which hooks it defines.
-3. The callback assigned by the plugin class to the hook is registered in a module variable.
-4. At the appropriate time (see below) the callbacks for a given hook are all called.
+From a user's perspective, plugins are enabled and disabled through the command
+line interface or through a UI. Users can also configure a plugin's behavior
+through the main Kolibri interface. See: :ref:`user-plugins`.
 
-A few notes:
 
-* Kolibri defines some hooks that are used in the core app. These can be found in the kolibri.plugins.registry module.
-* Arbitrary new hooks can be defined. At the Python level they're just dictionary keys. *Do not* user integers for
-  hook names -- those are reserved for the hooks defined by the Kolibri core. *Do* namespace your hooks!
-* *When* hook callbacks are called are entirely up to the hook definer. The core Kolibri app documents what hooks it
-  calls and when -- you can call core Kolibri hooks in your plugins too!
-* Callbacks *should* be called in the same order every time, but this is not a promise!
+.. note::
+    We have not yet written a configuration API, for now just make sure
+    configuration-related variables are kept in a central location of your
+    plugin.
+
+    It's up to the plugin to provide configuration ``Form`` classes and register
+    them.
```

Comment:
Yes it's a django.forms.Form instance that I'm referring to :)

Default behaviour could be to just serialize `forms.cleaned_data` and dump it in a file... but ultimately, it would be up to the plugin to define a call back, as with `form_valid()` on a FormView CBV.